scico/test/optimize/test_admm.py

```diff
@@ -145,6 +148,49 @@ def test_admm_generic(self):
         x = admm_.solve()
         assert (snp.linalg.norm(self.grdA(x) - self.grdb) / snp.linalg.norm(self.grdb)) < 1e-3
 
+    def test_admm_saveload(self):
+        maxiter = 5
+        x_ref = np.ones((16, 16), dtype=np.float32)
+        x_ref[4:-4, 4:-4] = 1.0
+        n = 3
+        psf = snp.ones((n, n), dtype=np.float32) / (n * n)
+        A = linop.CircularConvolve(h=psf, input_shape=x_ref.shape)
+        y = A(x_ref)
+        λ = 2e-2
+        ρ = 5e-1
+        f = loss.SquaredL2Loss(y=y, A=A)
+        g = λ * functional.L21Norm()
+        C = linop.FiniteDifference(x_ref.shape, circular=True)
+        admm0 = ADMM(
+            f=f,
+            g_list=[g],
+            C_list=[C],
+            rho_list=[ρ],
+            x0=A.adj(y),
+            maxiter=maxiter,
+            subproblem_solver=CircularConvolveSolver(),
+        )
+        admm0.solve()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            path = os.path.join(tmpdir, "admm.npz")
+            admm0.save_state(path)
+            admm0.solve()
```

Comment:
Why is `admm0.solve()` called twice?

Reply:
The sequence is as follows:

1. Solve 5 iterations of `admm0`.
2. Save `admm0` state.
3. Solve another 5 iterations of `admm0`.
4. Initialize `admm1` with state of `admm0` after 5 iterations.
5. Solve another 5 iterations of `admm1`.
6. Compare results for `admm0` and `admm`.

The idea is that the additional 5 iterations after saving the state allow the solutions to diverge if any component of the state is not properly saved, so that the final test can just be on the solution rather than on all state components.